### DISCUSSION TITLE
Poll: interest in deploying frontend websites?

### DISCUSSION THREAD

**Discussion by mnapoli** [serverless/serverless#11056] [Polls]
We want to evaluate the interest in deploying frontend websites (like React, Vue, etc.) natively with Serverless Framework.

The idea would be to deploy the infrastructure (S3 bucket + CloudFront + domain name) and upload the assets (JS, HTML, CSS files) via `serverless.yml` and the `serverless deploy` command.

If you want to provide more details, for example which framework you use (React, Vue or other?), how you are currently deploying it, etc, feel free to post a comment in this thread!

**Comment by chris-hinds** [comment]
I know this is totally out of scope but combine this with the ability to support custom/vanity domains (i.e. customer can create a subdomain of `foo.customer.com` and CNAME it to the deployed site at `sasscompany.com`) and you will have the ultimate stack for SASS companies and startups.

**Comment by jchou1271** [comment]
Native support for deploying nextjs website (static on S3 + SSR/ISR on lambda + cloudfront + domain name) would be ideal.

**Reply by cheru94** [reply]
Agreed ! 😄

**Reply by mnapoli** [reply]
SSR is definitely a bit more complex, I would be curious if a way to host the Next website statically at first (https://nextjs.org/docs/advanced-features/static-html-export) would be interesting to you?

**Reply by jchou1271** [reply]
Yes, of course I would be very interested in this as a first step, working towards a future state of static + dynamic rendering with SSR.  Thank you 🙏

**Reply by M1ke** [reply]
Just a note on this; we've been investigating SSR with Lambda. One issue is that you lose a lot of the advantage of how Next is _meant_ to work with SSR, whereby it's doing a lot of caching. Unless you do something like hook up EFS for file-level cache you lose a lot of the SSR benefits and may as well just be writing a SPA (which you can deploy statically) along with an API running on Lambda instead.

**Reply by mnapoli** [reply]
+1 @M1ke, this is the conclusion we came to in Lift as well.

I think https://open-next.js.org/ by the SST folks has the highest chance of success, worth watching probably.

**Comment by BenCoffeed** [comment]
webpack spa + bref backend ❤️

**Comment by luanmuniz** [comment]
I think this feature is in the scope of plugins instead of the native framework

There are too many ways to build and deploy frontend apps and too many ways to customize the installation with extra AWS services and configurations. I think this would add load to the maintainability of the framework unnecessarily and make other parts of the framework potentially more complicated.

I was using this for frontend apps until it went unmaintained https://github.com/stojanovic/scottyjs
It's not a small feature and it has its own set of bugs, especially if integrated directly into the core code.

You can have the better of both worlds and maintain an official frontend app plugin. I think this would be a better approach to adding this feature to the core. You can even ship the plugin with the framework as long as it's not "in the core code" and it's shipped as a plugin I think it would have fewer drawbacks.

**Comment by Markkos89** [comment]
I was just deploying my frontend apps with github actions syncronizing to S3 buckets. 

> The idea would be to deploy the infrastructure (S3 bucket + CloudFront + domain name) and upload the assets (JS, HTML, CSS files) via serverless.yml and the serverless deploy command.

i like it a lot !! Looking forward :)

**Comment by ennioVisco** [comment]
Perhaps unpop opinion, but what about joining forces with https://github.com/unjs/nitro? They have a lot of experience on frontend and typescript stuff, but clearly lack the ton of things you know on cloud-based stuff. 
Perhaps it could lead on a v4 of serverless far superior in performances :)
